### DISCUSSION TITLE
Proposal: more frequent z-releases for Argo CD operator

### DISCUSSION THREAD

**Discussion by jaideepr97** [argoproj-labs/argocd-operator#839] [Ideas]
As maintainers of this repository, the process we have been following until this point has been that we tend to keep accumulating bug fixes and other contributions within the master branch and tend to release them all together in the next minor release.

This results in relatively big gaps between releases, and bugs that have been fixed already don't see the light of day until months later sometimes, which ends up becoming inefficient from an end-user perspective. 

The maintainers of this project have discussed updating our processes to try and have more frequent z-releases for the operator so that bug fixes can be shipped out and made available more quickly to users. This would have the added benefit of greater feedback opportunities from the community to catch regressions etc, and would also shorten the window between bug identification and resolution.

One of the ideas we have to try to implement this is to use the PR prefixes currently in place. Bug fix PRs are prefixed with "fix" - one idea could be to cut a new z-release after every 3rd/4th PR containing this prefix that is merged, for example. Other ideas are welcome as well

On the other hand - one of the overheads of this I can see would be that we would have to branch out a lot sooner which would result in each fix needing to be committed to both master as well as the release branch 

Any inputs appreciated! TIA

**Comment by iam-veeramalla** [comment]
Thanks alot @jaideepr97 for starting the discussion. I agree that currently we almost do not deal with any z-stream releases. It is not a good practice. We should target a specific set of bugs and do frequent z-stream releases.
